### ISSUE TITLE
Question: Rotate jingle problem

### ISSUE THREAD

**Issue by sonyarianto** [savonet/liquidsoap#1484]
I am new to Liquidsoap and I am using Liquidsoap 1.4.3.

I have simple script like this that combine jingle playlist and songs playlist. My expecation is on every 3 songs will play 1 random jingle. Here is my script.

Songs folder `media-files` contains around 10 .flac files.
Jingle folder `id-files` contains 3 MPs
Output only to alsa since I just want to check the rotation.

```
songs = crossfade(playlist(mode="randomize", reload=10, reload_mode="watch", "media-files"))
jingles = playlist(mode="randomize", reload=10, reload_mode="watch", "id-files")

radio = rotate(weights=[1, 3], [jingles, songs])

output.alsa(fallible=true, radio)
```
The result above is the jingle only played on the beginning after run the script, after that the jingle never played again even after more that 5 songs.

What's wrong with my script above? I tried with other variatio with random also not working. Can someone help me?

Thanks in advance.

**Comment by martinkirch** [comment]
Hello,
I'm not sure, but it's possibly linked to #1468 (that will be fixed in 1.4.4).

You can try this method to play jingles regularly : 
```
jingles = delay(900.,
  playlist(mode="randomize", reload=10, reload_mode="watch", "id-files")
)

radio = fallback([jingles, songs])
```
The `delay` operator ensures that the wrapped `playlist` will not be available after playing a jingle, for 900 seconds. Meanwhile, the `fallback` operator will let `songs` play. This ensures that a jingle plays at most every 15 minutes.

**Comment by sonyarianto** [comment]
@martinkirch thank you, aha, I don't know that problem (#1468) yet. I will try your recipe above and will let you know the result.

**Comment by sonyarianto** [comment]
@martinkirch I tried above method and it doesn't work on me. I don't know why.

**Comment by DoM1niC** [comment]
> Hello,
> I'm not sure, but it's possibly linked to #1468 (that will be fixed in 1.4.4).
> 
> You can try this method to play jingles regularly :
> 
> ```
> jingles = delay(900.,
>   playlist(mode="randomize", reload=10, reload_mode="watch", "id-files")
> )
> 
> radio = fallback([jingles, songs])
> ```
> 
> The `delay` operator ensures that the wrapped `playlist` will not be available after playing a jingle, for 900 seconds. Meanwhile, the `fallback` operator will let `songs` play. This ensures that a jingle plays at most every 15 minutes.

I use 1.4.4 my problem is similar here my short jingles will play twice instead of one only but still different, a long Jingle works fine ... this workaround will fix my problem too?

**Comment by martinkirch** [comment]
Maybe, give it a try and let us know

**Comment by smimram** [comment]
I can confirm that this is broken in 1.4.3
```
songs   = sine(duration=1., 440.)
jingles = sine(duration=1., 880.)
radio   = rotate(weights=[1, 3], [jingles, songs])
output.alsa(fallible=true, radio)
```
and that this is working in 2.0
```
songs   = single("synth:frequency=440,d=1")
jingles = single("synth:frequency=880,d=1")
radio   = rotate(weights=[1, 3], [jingles, songs])
output.alsa(fallible=true, radio)
```

**Comment by sonyarianto** [comment]
@smimram wew finally there is confirmation, thank you. Now I am using 1.4.4 (on Opam) and without short jingles hahah (because of above problems). Will try again on 2.0.0. When the 2.0.0 will available? I saw on Docker version already use 2.0.0 ya?

Anyway thanks again for the reply. It helps a lot :) I appreciate your time and efforts.
